Backpack-SwiftUI/RatingBar/Classes/BPKRatingBar.swift

```diff
@@ -0,0 +1,146 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2022 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+public struct BPKRatingBar: View {
+    private let label: String
+    private let rating: Float
+    private let accessibilityLabel: String
+    private let ratingScale: RatingScale
+    private let showScale: Bool
+    private let style: Style
+    
+    /// Initializes a new instance of the `BPKRatingBar` struct.
+    /// - Parameters:
+    ///   - label: The label for the rating bar.
+    ///   - rating: The rating value.
+    ///   - accessibilityLabel: The accessibility label for the rating bar.
+    ///   - ratingScale: The rating scale for the rating bar. Default value is `.zeroToFive`.
+    ///   - showScale: A boolean value indicating whether to show the rating scale. Default value is `true`.
+    ///   - style: The style of the rating bar. Default value is `.default`.
+    public init(
+        label: String,
+        rating: Float,
+        accessibilityLabel: String,
+        ratingScale: RatingScale = .zeroToFive,
+        showScale: Bool = true,
+        style: Style = .default
+    ) {
+        self.label = label
+        self.rating = rating
```

Comment:
We probably want to add some clamping to keep the rating between the scale? I saw the tests with a rating of -2.5 or 6.0/5 and they don't make sense from a customer perspective

Reply:
I saw this, but ended up not handling it within backpack to let this logic be in the client.
I will handle it within backpack now though as I see the value having it!